configuration/src/configuration.ts

```diff
@@ -186,6 +204,42 @@ export class Configuration implements IConfiguration {
         }
         return null;
     }
+
+    /**
+     * Gets the secret value associated with the specified key from the
+     * ordered configuration providers. For example, with a configuration like:
+     * { "AAD_SECRET": { "id": "https://foo.vault.azure.net/secrets/aad-secret"} },
+     * config.getSecret('AAD_SECRET') will fetch the value of the secret with the
+     * specified id from KeyVault.
+     * If the object already contains a 'value' field, this method will return
+     * the object as-is (useful for debugging purposes.)
+     *
+     * @param {string | string[]} key - Name of the variable to get.
+     * @return {Promise<Secret>} Value of the secret located at the id specified
+     * in the configuration object named by key. Null if no value is set.
+     */
+    public getSecret(key: string | string[]): Promise<Secret> {
+        const secret = this.get<Secret>(key);
+        if (secret === null) {

```

Comment:
This is the convention used by the other config apis. Hunter standardized on null to denote the non-existence of keys: undefined and other falsey values are passed through as-is by the other APIs, so I wanted to keep it consistent here as well.